src/path.ts

```diff
@@ -17,7 +17,7 @@ const _ROOT_FOLDER_RE = /^\/([A-Za-z]:)?$/;
 
 // Force POSIX contants
 export const sep = "/";
-export const delimiter = ":";
+export const delimiter = /^win/i.test(process.platform) ? ";" : ":";
```

Comment:
pathe is designed to have consistent platfrom-independent behavior, while `delimiter` is (rightfully) useful for parsing a platform-dependent value, as a compromise, i think we can also think about setting this export to a regex which is accepted by `String.prototype.split` still

```suggestion
export const delimiter = /[:;]/
```